src/System.Text.RegularExpressions/tests/Performance/Perf.Regex.cs

```diff
@@ -0,0 +1,442 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+using Microsoft.Xunit.Performance;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    /// <summary>
+    /// Performance tests for Regular Expressions
+    /// </summary>
+    public class Perf_Regex
+    {
+        private const int innerIterations = 100;
+
+        [Benchmark]
+        [MeasureGCAllocations]
+        public void Match()
+        {
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < innerIterations; i++)
+                    {
+                        foreach(var test in Match_TestData())
```

Comment:
There's work here unrelated to Regex.Match that it would be nice to move out of the measurement, e.g. store the data from Match_TestData() into an array at the beginning of the method once and then access the array here.